bson/src/main/org/bson/io/BasicOutputBuffer.java

```diff
@@ -131,7 +132,7 @@ public void truncateToPosition(final int newPosition) {
     @Override
     public List<ByteBuf> getByteBuffers() {
         ensureOpen();
-        return Arrays.<ByteBuf>asList(new ByteBufNIO(ByteBuffer.wrap(buffer, 0, position).duplicate()));
+        return Arrays.<ByteBuf>asList(new ByteBufNIO(ByteBuffer.wrap(buffer, 0, position).duplicate().order(LITTLE_ENDIAN)));
```

Comment:
It's necessary because there's now a consumer that cares about the endianness of the byte buffers.  Previously, all consumers treated this as a raw byte array but now CompressedMessage calls ByteBuf.getInt, which is endian-sensitive.  

While we could change the endian-ness in CompressedMessage, it seems safer to do it in OutputBuffer, since BSON is always little-endian.

As for backwards compatibility, I'm not too concerned, as:

* the endian-ness was previously unspecified
* i can't see why anyone would ever want it to be big-endian, given that BSON is little-endian
* all current known use cases don't care about endian-ness
* it's unlikely anyone else is using this method